### DISCUSSION TITLE
Simultaneously presenting audio and text

### DISCUSSION THREAD

**Discussion by kevinvschie** [jspsych/jsPsych#1080] [Q&A]
I'm just starting out with jsPsych so am looking for some advice. What I want to achieve is the following: I want to present participants with instructions (sentence by sentence on screen) and want each sentence to be accompanied by an audio fragment (essentially reading the sentence, while participants read along). How can I best achieve this? Perhaps an audio-keyboard-response plugin and put the written sentence in the prompt? Are there better ways to achieve this?

**Comment by jodeleeuw** [comment]
> Perhaps an audio-keyboard-response plugin and put the written sentence in the prompt

That's exactly what I would do!

**Reply by kevinvschie** [reply]
Where would I need to store the audio files? Do I need to map a subfolder or can I put the audio files in the same folder as where the html is? Also, what file format does it need to be? Mp3?

**Reply by kevinvschie** [reply]
I figured this simple script should work (but it doesn't). I believe there is no need to preload the audiofile. I created a folder 'Audio' which is in the jspsych-6.1.0 folder. Perhaps that shouldn't be there? I tried a couple of other things, but that did not change anything. 

I keep finding a bunch of erros in the developer tools, but the primary one being: "jspsych.js:44 The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page.".  I have attached the script as a text (couldn't get it to load properly here). if anyone has any suggestions, I'd gladly hear them!

[PocketTNT_Instructions.txt](https://github.com/jspsych/jsPsych/files/5303801/PocketTNT_Instructions.txt)

**Reply by becky-gilbert** [reply]
Hi @kevinvschie! 

> Where would I need to store the audio files? Do I need to map a subfolder or can I put the audio files in the same folder as where the html is? Also, what file format does it need to be? Mp3?

Yep you can store your audio files anywhere in your experiment directory - in the main folder with your HTML file or in a subfolder. I would suggest using mp3 if possible. You can also use wav files but they're larger so could cause loading delays if you're using many and/or big files.

Your script looks good to me. The "AudioContext was not allowed to start" error is nothing to worry about - it's just a warning that you'll see whenever you use audio. I'm guessing that you're getting the [CORS errors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) that occur when using the Web Audio API and opening the HTML file directly in the browser (compared to running it on a server). Once your file is on a server, you won't get this error anymore. But in the meantime, the fix for this is to add `use_webaudio: false` to your `jsPsych.init` settings. Sorry that the [documentation on this issue](https://www.jspsych.org/overview/experiment-options/#choose-the-method-for-playing-audio-files) isn't easy to find... In the future I think we need to add a dedicated page for audio-related issues.
